client/signup/steps/page-picker/use-cart-for-difm.tsx

```diff
@@ -0,0 +1,292 @@
+						product.quantity === 1 ? translate( 'Extra Page' ) : translate( 'Extra Pages' )
+					}`,
+					productOriginalName: product.product_name,
+					lineCost: product.cost,
+					productCost: product.cost,
+					meta: product.item_original_cost_for_quantity_one_display + ' ' + translate( 'Per Page' ),
+				};
+
+			default:
+				return {
+					productSlug: product.product_slug,
+					nameOverride: 'Website Design Service',
+					productOriginalName: product.product_name,
+					lineCost: product.cost,
+					productCost: product.cost,
+					meta: translate( 'One-time fee' ),
+				};
+		}
+	} );
+
+	// Add a 0 line item product if there is no extra page in the cart
+	if ( ! cartItems.some( ( c ) => c.productSlug === WPCOM_DIFM_EXTRA_PAGE ) ) {
+		cartItems.push( {
+			productSlug: extraPageProduct.product_slug,
+			productOriginalName: extraPageProduct.product_name,
+			lineCost: 0,
+			productCost: 0,
+			nameOverride: `0 ${ translate( 'Extra Pages' ) }`,
+			meta: translate( '%(perPageCost)s Per Page', {
+				args: {
+					perPageCost: formatCurrency( extraPageProduct.cost, currencyCode, { precision: 0 } ),
+				},
+			} ),
+		} );
+	}
+	return cartItems;
+}
+
+const debounce = ( callback: ( ...args: any[] ) => any, timeout: number ) => {
+	let timeoutId: number | undefined = undefined;
+	return ( ...args: any[] ) => {
+		window.clearTimeout( timeoutId );
+		timeoutId = window.setTimeout( () => {
+			callback( ...args );
+		}, timeout );
+	};
+};
+
+export function useCartForDIFM( selectedPages: string[] ): {
+	items: CartItem[];
+	total: string | null;
+	isLoading: boolean;
+	isPendingUpdate: boolean;
+	isCartUpdateStarted: boolean;
+} {
+	//This state is used by loader states to provide immediate feedback when a deebounced change happens to a cart
+	const [ isCartUpdateStarted, setIsCartUpdateStarted ] = useState( false );
+
+	const signupDependencies = useSelector( getSignupDependencyStore );
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const proPlan = useSelector( ( state ) => getProductBySlug( state, PLAN_WPCOM_PRO ) );
+	const premiumPlan = useSelector( ( state ) => getProductBySlug( state, PLAN_PREMIUM ) );
+	const activePremiumPlanScheme = isEnabled( 'plans/pro-plan' ) ? proPlan : premiumPlan;
+	const { newOrExistingSiteChoice, siteId, siteSlug } = signupDependencies;
+
+	const extraPageProduct = useSelector( ( state ) =>
+		getProductBySlug( state, WPCOM_DIFM_EXTRA_PAGE )
+	);
+	const difmLiteProduct = useSelector( ( state ) => getProductBySlug( state, WPCOM_DIFM_LITE ) );
+
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+
+	const cartKey = useSelector( ( state ) => getSiteId( state, siteSlug ?? siteId ) );
+	const { replaceProductsInCart, responseCart, isLoading, isPendingUpdate } = useShoppingCart(
+		cartKey ?? undefined
+	);
+
+	const serializedDifmProduct = JSON.stringify( difmLiteProduct );
+	const serializedSignupDependencies = JSON.stringify( signupDependencies );
```

Comment:
Seems like there is a change in the reference to the object (Object seems to be recreated somewhere along the state). Because without it I am met with a re-render loop on the useEffect. I will explore this a bit more but for now this seems like the most sensible approach for this.